pkg/machinery/interfaces.go

```diff
@@ -44,6 +44,10 @@ type Template interface {
 	GetBody() string
 	// SetTemplateDefaults sets the default values for templates
```

Comment:
@ruanxin, 

Could you please fix the CI issues so we are able to move forward? 

a) Lint: 

  Error: SA4005: ineffective assignment to field fakeTemplate.parseDelimLeft (staticcheck)
[27](https://github.com/kubernetes-sigs/kubebuilder/runs/8119443645?check_suite_focus=true#step:5:29)
  Error: SA4005: ineffective assignment to field fakeTemplate.parseDelimRight (staticcheck)
[2](https://github.com/kubernetes-sigs/kubebuilder/runs/8119443645?check_suite_focus=true#step:5:30)  
[29](https://github.com/kubernetes-sigs/kubebuilder/runs/8119443645?check_suite_focus=true#step:5:31)
  Error: issues found
[30](https://github.com/kubernetes-sigs/kubebuilder/runs/8119443645?check_suite_focus=true#step:5:32)
  Ran golangci-lint in 18854ms

Note: You can run make lint locally to check it out

b) The API one we just need to check / ensure that it does not break anything and we can move forward with it failing

c) the e2e test failures we need to ensure that are not related to this one

d) the coveralls we need to ensure that we add tests to pass on this one.

Could you please also rebase with the master to ensure that you have the latest changes?

Reply:
Hi @camilamacedo86 ,

I resolved this lint complain, it requires some refactoring. Just explain here, the `fakeTemplate` have to use pointer receiver, actually it's a golang best practice [If the method needs to mutate the receiver, the receiver must be a pointer.](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type), because of this, the struct has to be allocated as `&fakeTemplate` since the receiver is pointer `*fakeTemplate` now, all basic golang stuff, just want to clarify the change.

btw, I also rebased with the master.